atc/creds/keyvault/reader.go

```diff
@@ -0,0 +1,111 @@
+package keyvault
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// ErrSecretNotFound is the error returned when a secret cannot be found by the
+// reader
+var ErrSecretNotFound = fmt.Errorf("secret not found")
+
+// SecretReader is an interface that can get and list secrets from a store.
+// This is needed here because there are no fakes provided with the Azure SDK
+// for Go
+type SecretReader interface {
+	Get(string) (string, bool, error)
+	List(string) ([]string, error)
+}
+
+type keyVaultReader struct {
+	client      keyvault.BaseClient
+	keyVaultURL string
+}
+
+// NewKeyVaultReader returns a keyvault implementation of a SecretReader. It
+// requires a fully configured Key Vault client and the URL of the Key Vault
+func NewKeyVaultReader(client keyvault.BaseClient, keyVaultURL string) SecretReader {
+	return &keyVaultReader{
+		client:      client,
+		keyVaultURL: keyVaultURL,
+	}
+}
+
+// Get returns the value of the given secret name, a bool if it was found, and
+// an error if there was an issue fetching the secret
+func (k *keyVaultReader) Get(name string) (string, bool, error) {
+	// Right now we aren't timing out, we may want to add a configurable option
```

Comment:
Makes sense. I'm OK with this as-is for now. This may be something to consider generically rather than having each credential manager implement their own timeout flags.

Made a note for this here: https://github.com/concourse/rfcs/issues/5#issuecomment-472427713